recorder/src/main/cpp/profile_writer.hh

```diff
@@ -44,5 +47,65 @@ public:
     void write_header(const recording::RecordingHeader& rh);
 
     void append_wse(const recording::Wse& e);
+
+    void flush();
+};
+
+typedef std::uint32_t FlushCtr;
+
+struct SerializationFlushThresholds {
+    FlushCtr cpu_samples;
+
+    SerializationFlushThresholds() : cpu_samples(100) {}
```

Comment:
I didn't think it was necessary in the first cut, but we can make it configurable if necessary. Created an issue so we can do this later: https://github.com/Flipkart/fk-prof/issues/33